doc/source/ray-core/examples/batch_training.ipynb

```diff
@@ -254,125 +292,70 @@
-    "            \"pickup_at\",\n",
-    "            \"dropoff_at\",\n",
-    "            \"pickup_location_id\",\n",
-    "            \"dropoff_location_id\",\n",
-    "        ],\n",
-    "    ).to_pandas()"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "The `task` Ray task contains all logic necessary to load a data batch, transform it and fit and evaluate models on it.\n",
     "\n",
-    "You may notice that we have previously defined `fit_and_score_sklearn` as a Ray task as well and set it to be executed from inside `task`. This allows us to dynamically create a [tree of tasks](task-pattern-tree-of-tasks), ensuring that the cluster resources are fully utillized. Without this pattern, each `task` would need to be assigned several CPU cores for the model fitting, meaning that if certain models finish faster, then those CPU cores would stil stay occupied. Thankfully, Ray is able to deal with nested parallelism in tasks without the need for any extra logic, allowing us to simplify the code."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 9,
-   "metadata": {},
-   "outputs": [],
-   "source": [
     "@ray.remote\n",
-    "def task(\n",
-    "    data: Union[str, pd.DataFrame],\n",
+    "def train_and_evaluate(\n",
     "    file_name: str,\n",
-    "    i: int,\n",
+    "    location_id: int,\n",
     "    models: List[BaseEstimator],\n",
-    "    load_data_func: Optional[Callable] = None,\n",
-    ") -> List[Tuple[BaseEstimator, float]]:\n",
-    "    if load_data_func:\n",
-    "        start_time = time.time()\n",
-    "        data = load_data_func(data, i)\n",
-    "        data_loading_time = time.time() - start_time\n",
-    "        print_time(f\"Data loading time for LocID: {i}: {data_loading_time}\")\n",
-    "\n",
+    ") -> Tuple[str, str, List[Tuple[BaseEstimator, float]]]:\n",
     "    # Cast PyArrow scalar to Python if needed.\n",
     "    try:\n",
-    "        i = i.as_py()\n",
+    "        location_id = location_id.as_py()\n",
     "    except Exception:\n",
     "        pass\n",
     "\n",
+    "    start_time = time.time()\n",
+    "    data = read_data(file_name, location_id)\n",
+    "    data_loading_time = time.time() - start_time\n",
+    "    print_time(f\"Data loading time for LocID: {location_id}: {data_loading_time}\")\n",
+    "\n",
     "    # Perform transformation\n",
     "    start_time = time.time()\n",
     "    data = transform_batch(data)\n",
     "    transform_time = time.time() - start_time\n",
-    "    print_time(f\"Data transform time for LocID: {i}: {transform_time}\")\n",
+    "    print_time(f\"Data transform time for LocID: {location_id}: {transform_time}\")\n",
     "\n",
-    "    return file_name, i, train_and_evaluate(data, models, i)"
+    "    return (\n",
+    "        file_name,\n",
+    "        location_id,\n",
+    "        train_and_evaluate_internal(data, models, location_id),\n",
+    "    )"
```

Comment:
For consistency with the above steps, how about extracting the timing steps into this function?

Something like:
```python
start_time = time.time()
results = train_and_evaluate_internal(data, models, location_id)
train_and_evaluate_time = time.time() - start_time
print_time(...)

return (file_name, location_id, results)
```